firmware/baseband/proc_btlerx.cpp

```diff
@@ -26,245 +27,476 @@
 
 #include "event_m4.hpp"
 
+uint32_t BTLERxProcessor::crc_init_reorder(uint32_t crc_init) {
+    int i;
+    uint32_t crc_init_tmp, crc_init_input, crc_init_input_tmp;
+
+    crc_init_input_tmp = crc_init;
+    crc_init_input = 0;
+
+    crc_init_input = ((crc_init_input) | (crc_init_input_tmp & 0xFF));
+
+    crc_init_input_tmp = (crc_init_input_tmp >> 8);
+    crc_init_input = ((crc_init_input << 8) | (crc_init_input_tmp & 0xFF));
+
+    crc_init_input_tmp = (crc_init_input_tmp >> 8);
+    crc_init_input = ((crc_init_input << 8) | (crc_init_input_tmp & 0xFF));
+
+    crc_init_input = (crc_init_input << 1);
+    crc_init_tmp = 0;
+
+    for (i = 0; i < 24; i++) {
+        crc_init_input = (crc_init_input >> 1);
+        crc_init_tmp = ((crc_init_tmp << 1) | (crc_init_input & 0x01));
+    }
+
+    return (crc_init_tmp);
+}
+
+uint_fast32_t BTLERxProcessor::crc_update(uint_fast32_t crc, const void* data, size_t data_len) {
+    const unsigned char* d = (const unsigned char*)data;
+    unsigned int tbl_idx;
+
+    while (data_len--) {
+        tbl_idx = (crc ^ *d) & 0xff;
+        crc = (crc_table[tbl_idx] ^ (crc >> 8)) & 0xffffff;
+
+        d++;
+    }
+
+    return crc & 0xffffff;
+}
+
+uint_fast32_t BTLERxProcessor::crc24_byte(uint8_t* byte_in, int num_byte, uint32_t init_hex) {
+    uint_fast32_t crc = init_hex;
+
+    crc = crc_update(crc, byte_in, num_byte);
+
+    return (crc);
+}
+
+bool BTLERxProcessor::crc_check(uint8_t* tmp_byte, int body_len, uint32_t crc_init) {
+    int crc24_checksum;
+
+    crc24_checksum = crc24_byte(tmp_byte, body_len, crc_init);  // 0x555555 --> 0xaaaaaa. maybe because byte order
+    checksumReceived = 0;
+    checksumReceived = ((checksumReceived << 8) | tmp_byte[body_len + 2]);
+    checksumReceived = ((checksumReceived << 8) | tmp_byte[body_len + 1]);
+    checksumReceived = ((checksumReceived << 8) | tmp_byte[body_len + 0]);
+
+    return (crc24_checksum != checksumReceived);
+}
+
+void BTLERxProcessor::scramble_byte(uint8_t* byte_in, int num_byte, const uint8_t* scramble_table_byte, uint8_t* byte_out) {
+    int i;
+
+    for (i = 0; i < num_byte; i++) {
+        byte_out[i] = byte_in[i] ^ scramble_table_byte[i];
+    }
+}
+
+// void BTLERxProcessor::demod_byte(int num_byte, uint8_t *out_byte)
```

Comment:
nit: unless you plan to come back to this, just delete dead code. It's always in the history if we need it.

Reply:
I actually did plan on it, but I don't need it now. Should I delete it even if I still plan to in the future refactoring?

Reply:
No, if you want it, keep it. But don't keep around dead code if it's really dead.